src/Modules/ExteriorPowers/Generic.jl

```diff
@@ -0,0 +1,184 @@
+  end
+  return (false, M, 0)
+end
+
+# Printing of exterior powers
+function show_exterior_product(io::IO, M::ModuleFP)
+  success, F, p = is_exterior_power(M)
+  success || error("module is not an exterior power")
+  print(io, "⋀^$p($F)")
+end
+
+function show_exterior_product(io::IO, ::MIME"text/html", M::ModuleFP)
+  success, F, p = is_exterior_power(M)
+  success || error("module is not an exterior power")
+  io = IOContext(io, :compact => true)
+  print(io, "⋀^$p$F")
+end
+
+function multiplication_map(M::ModuleFP)
+  has_attribute(M, :multiplication_map) || error("module is not an exterior power")
+  return get_attribute(M, :multiplication_map)::Map
+end
+
+function wedge_pure_function(M::ModuleFP)
+  has_attribute(M, :wedge_pure_function) || error("module is not an exterior power")
+  return get_attribute(M, :wedge_pure_function)::Map
+end
+
+function wedge_generator_decompose_function(M::ModuleFP)
+  has_attribute(M, :wedge_generator_decompose_function) || error("module is not an exterior power")
+  return get_attribute(M, :wedge_generator_decompose_function)::Map
+end
+
+# Given two exterior powers F = ⋀ ᵖM and G = ⋀ ʳM and an element 
+# v ∈ ⋀ ʳ⁻ᵖ M this constructs the module homomorphism associated 
+# to 
+#
+#   v ∧ - : F → G,  u ↦ v ∧ u.
+#
+# We also allow v ∈ M considered as ⋀ ¹M and the same holds in 
+# the cases p = 1 and r = 1.
+function wedge_multiplication_map(F::ModuleFP, G::ModuleFP, v::ModuleFPElem)
+  success, orig_mod, p = is_exterior_power(F)
+  if !success 
+    Fwedge1, _ = exterior_power(F, 1)
+    id = hom(F, Fwedge1, gens(Fwedge1))
+    tmp = wedge_multiplication_map(Fwedge1, G, v)
+    return compose(id, tmp)
+  end
+
+  success, orig_mod_2, q = is_exterior_power(G)
+  if !success
+    Gwedge1, _ = exterior_power(G, 1)
+    id = hom(Gwedge1, G, gens(G))
+    tmp = wedge_multiplication_map(F, Gwedge1, v)
+    return compose(tmp, id)
+  end
+
+  orig_mod === orig_mod_2 || error("modules must be exterior powers of the same module")
+  H = parent(v)
+
+  # In case v comes from the original module, convert.
+  if H === orig_mod
+    M, _ = exterior_power(orig_mod, 1)
+    w = M(coordinates(v))
+    return wedge_multiplication_map(F, G, w)
+  end
+
+  success, orig_mod_2, r = is_exterior_power(H)
+  success || error("element is not an exterior product")
+  orig_mod_2 === orig_mod || error("element is not an exterior product for the correct module")
+  p + r == q || error("powers are incompatible")
+  
+  # map the generators
+  img_gens = [wedge(v, e, parent=G) for e in gens(F)]
+  return hom(F, G, img_gens)
+end
+
+# The wedge product of two or more elements.
+function wedge(u::ModuleFPElem, v::ModuleFPElem; 
+    parent::ModuleFP=begin
+      success, F, p = is_exterior_power(Oscar.parent(u))
+      if !success 
+        F = Oscar.parent(u)
+        p = 1
+      end
+      success, _, q = is_exterior_power(Oscar.parent(v))
+      !success && (q = 1)
+      exterior_power(F, p + q)[1]
+    end
```

Comment:
How else to do it? I know it's hard to read, but I would have to write an internal initialization function otherwise. Would that make it clearer? Either way: That part has already been merged. So any discussion is about "how to do it next time".